src/stores/JourneyStore.js

```diff
@@ -1,19 +1,29 @@
 import {extendObservable, action} from "mobx";
 import {pick} from "lodash";
+import getJourneyId from "../helpers/getJourneyId";
 
 export default (state) => {
   extendObservable(state, {
     selectedJourney: null,
   });
 
   const setSelectedJourney = action((journey = null) => {
-    state.selectedJourney = pick(
-      journey,
-      "jrn",
-      "oday",
-      "uniqueVehicleId",
-      "journeyStartTime"
-    );
+    if (
+      !journey ||
+      (state.selectedJourney &&
+        getJourneyId(state.selectedJourney) === getJourneyId(journey))
+    ) {
+      state.selectedJourney = null;
+    } else {
+      state.selectedJourney = pick(
+        journey,
+        "oday",
+        "uniqueVehicleId",
+        "journeyStartTime",
+        "directionId",
+        "routeId"
+      );
+    }
```

Comment:
This action acts as a toggle. If called with a journey that is already selected, it will be unselected.